### PR TITLE
align properties better with our existing integrations;  Add a few ne…

### PR DESCRIPTION
**Pull request by gcooney**
Couple main things here:

1)  Remove the checked in compiled typescript and node modules.  Add scripts and docs around only checking these on release branches
2)  Standardize property names with azure devops pipelines
3)  Add PR information when available

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
   "main": "lib/index.js",
   "scripts": {
     "test": "tsc && mocha ./tests/_suite.js",
-    "build": "tsc"
+    "build": "tsc",
+    "release": "npm install && npm run build && npm prune --production && git add -f node_modules/* && git add -f lib/*"
```
Comment (gcooney) [inline package.json]: script most of release process to simplify

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,27 +1,21 @@
...
 
-Actions need the compiled code checked in. This means your changes will only
-take effect if you run the build to generate .js files from the .ts Typescript
-files and check the .js files in as well.
-
-### Adding new dependencies
-
-Check in the Action with everything it needs to run. If you update dependencies
-you will need to commit the changes to Node modules as well.
+Releases are done of of release branches. Each major release version should get
```
Comment (simonychoy) [inline CONTRIBUTING.md]: `of of` should be `on`? I would suggest rewording this to avoid using the word `done` because it can ambiguous as to what that means.

File: src/entities/Deployment.ts [hunk 3]
```diff
@@ -15,9 +15,23 @@ interface TriggeredPlanSummary {
 }
 
 export interface DeploymentProperties {
-  committer: string | undefined;
-  repositoryAction: string | undefined;
-  repositoryBranchName: string | undefined;
-  repositoryRevisionNumber: string | undefined;
```
Comment (simonychoy) [inline src/entities/Deployment.ts]: Was the revision number the commit hash or something? And does it match to one of the properties below?

File: src/entities/Deployment.ts [hunk 4]
```diff
@@ -15,9 +15,23 @@ interface TriggeredPlanSummary {
...
+  repository_commit_username: string | undefined;
+  repository_action: string | undefined;
+  repository_name: string | undefined;
+  repository_branch_name: string | undefined;
+  repository_url: string | undefined;
+  repository_pull_request_url?: string | undefined;
+  repository_pull_request_number?: number | undefined;
+  repository_pull_request_title?: string | undefined;
+  repository_pull_request_merged_at?: string | undefined;
+  repository_pull_request_created_at?: string | undefined;
```
Comment (simonychoy) [inline src/entities/Deployment.ts]: Is there a reason not to use the `PullRequest` type instead?

Comment (gcooney) [reply src/entities/Deployment.ts]: I think the api properties bundle is currently looking for string -> string.

Comment (gcooney) [reply src/entities/Deployment.ts]: I initially had it that way but was getting 500s

File: src/entities/Deployment.ts (hunk 3 above)
Comment (gcooney) [reply src/entities/Deployment.ts]: its mapped to the top level 'revision' instead of being in the properties map.  There is a value we support in the properties map as well but I'm nor sure we need to double send it.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,27 +1,21 @@
...
 
-Actions need the compiled code checked in. This means your changes will only
-take effect if you run the build to generate .js files from the .ts Typescript
-files and check the .js files in as well.
-
-### Adding new dependencies
-
-Check in the Action with everything it needs to run. If you update dependencies
-you will need to commit the changes to Node modules as well.
+Releases are build on release branches. Each major release version should get
```
Comment (simonychoy) [inline CONTRIBUTING.md]: sorry, nit: should be `built`

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,27 +1,21 @@
...
+its own branch, e.g. release_v1.  
+Releases can then be done from that branch by running
```
Comment (simonychoy) [inline CONTRIBUTING.md]: should `done` here be triggered?
